app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -509,6 +554,102 @@ public void onTabReselected(TabLayout.Tab tab) {
+                    return;
+                }
+                BarcodeValues barcodeValues = Utils.parseSetBarcodeActivityResult(Utils.BARCODE_SCAN, result.getResultCode(), intent, getApplicationContext());
+
+                cardId = barcodeValues.content();
+                barcodeType = barcodeValues.format();
+                barcodeId = "";
+            }
+        });
+
+        mCropperLauncher = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
+            Intent intent = result.getData();
+            if (intent == null){
+                Log.d("cropper", "ucrop returned a null intent");
+                return;
+            }
+            if (result.getResultCode() == Activity.RESULT_OK) {
+                Uri debugUri = UCrop.getOutput(intent);
+                if (debugUri == null){
+                    throw new RuntimeException("ucrop returned success but not destination uri!");
+                }
+                Log.d("cropper", "ucrop produced image at " + debugUri);
+                Bitmap bitmap = BitmapFactory.decodeFile(getCacheDir() + "/" + TEMP_CROP_IMAGE_NAME);
+
+                if (bitmap != null) {
+                    bitmap = Utils.resizeBitmap(bitmap);
+                    if (requestedFrontImage()) {
+                        mFrontImageUnsaved = true;
+                        setCardImage(cardImageFront, bitmap);
+                    } else {
+                        mBackImageUnsaved = true;
+                        setCardImage(cardImageBack, bitmap);
+                    }
+                    Log.d("cropper", "mRequestedImage: " + mRequestedImage);
+                    mCropperFinishedType = mRequestedImage;
+                    hasChanged = true;
+                } else {
+                    Toast.makeText(LoyaltyCardEditActivity.this, R.string.errorReadingImage, Toast.LENGTH_LONG).show();
+                }
+            }else if(result.getResultCode() == UCrop.RESULT_ERROR){
+                Throwable e = UCrop.getError(intent);
+                if (e == null){
+                    throw new RuntimeException("ucrop returned error state but not and error!");
+                }
+                Log.e("cropper error", e.toString());
+            }
+        });
+
+        mCropperOptions = new UCrop.Options();
+        setCropperOptions();
+        setCropperTheme();
+    }
+
+    private void setCropperOptions(){
+        mCropperOptions.setCompressionFormat(TEMP_CROP_IMAGE_FORMAT);
+        mCropperOptions.setFreeStyleCropEnabled(true);
+        mCropperOptions.setHideBottomControls(false);
+        // needed when bottom controls are hidden
+        //mCropperOptions.setAllowedGestures(UCropActivity.ALL, UCropActivity.ALL, UCropActivity.ALL);
+        mCropperOptions.setAspectRatioOptions(0,
+                new AspectRatio(null, 1, 1),
+                new AspectRatio(getResources().getString(R.string.card),(float)85.6,(float)53.98 )
```

Comment:
Instead of casting this to a float, using `85.6f` and `53.98f` as values is probably cleaner and more readable.